libraries/base/Data/Monoid.hs

```diff
@@ -16,8 +16,29 @@
 -- Stability   :  experimental
 -- Portability :  portable
 --
--- A class for monoids (types with an associative binary operation that
--- has an identity) with various general-purpose instances.
+-- A type @a@ is a 'Monoid' if it provides an associative function ('<>')
+-- that lets you combine any two values of type @a@ into one, and a neutral
+-- element (`mempty`) such that 
+--
+-- > a <> mempty == mempty a == a
+--
+-- ==== __Examples__
+-- >>> Sum 1 <> Sum 2 <> Sum 3 <> Sum 4 :: Sum Int
```

Comment:
@Daniel-Diaz how about now? I start of both scenarios (Sum and Product) with the invoking `<>` first, then use `mconcat`. I believe this should make it easier to deduce what is going on even without having previously seen `mconcat`.